src/btf.h

```diff
@@ -124,10 +122,13 @@ class BTF {
                                  std::unordered_set<std::string>& types) const;
   std::string get_all_funcs_from_btf(const BTFObj& btf_obj) const;
   std::string get_all_raw_tracepoints_from_btf(const BTFObj& btf_obj) const;
-  std::map<std::string, std::vector<std::string>> get_params_from_btf(
+  FuncParamLists get_params_impl(const std::set<std::string>& funcs,
+                                 bool is_rawtracepoints) const;
```

Comment:
I don't want to over-complicate things but how about we pass a callback (either `get_params` or `get_rawtracepoint_params`) instead of `is_tracepoints` here?

Reply:
Good idea. Done.